### PR TITLE
Add S3 Uploader

### LINKED ISSUE #61: [Suggestion] S3 Support
Adding s3 support would be quite nice for us since we have a bit of developers who have access to the files which means they would be able to see SFTP creds if we use sftp/ftp

### PR DESCRIPTION
**Pull request by vpzomtrrfrt**
Should resolve #61 (though it seems to have already been closed?)

### CODE REVIEW

**Review by MaxMaeder — APPROVED** [review]
LGTM!! Thanks for contributing!

**Comment by rhld16** [comment]
Two minor changes requested, but other than that - looks great! Thanks for contributing 😃

**Comment by vpzomtrrfrt** [comment]
What would those changes be?

**Comment by rhld16** [comment]
@vpzomtrrfrt Sorry for the delay, take a look at my review to see the requested changes

**Comment by vpzomtrrfrt** [comment]
I don't see a review from you

File: DriveBackup/src/main/java/ratismal/drivebackup/TestThread.java [hunk 1]
```diff
@@ -144,6 +145,14 @@ private void testUploadMethod(String testFileName, int testFileSize, String meth
                     return;
                 }
                 break;
+            case "s3":
+                if (config.backupMethods.s3.enabled) {
+                    uploadMethod = new S3Uploader(logger, config.backupMethods.s3);
+                } else {
+                    sendMethodDisabled(logger, NextcloudUploader.UPLOADER_NAME);
```
Comment (lrmtheboss) [inline DriveBackup/src/main/java/ratismal/drivebackup/TestThread.java:152]: Should be using S3Uploader instead of NextcloudUploader

**Review by lrmtheboss — DISMISSED** [review]
Just a couple of small changes

**Review by lrmtheboss — CHANGES_REQUESTED** [review]
Just a couple of small changes

File: DriveBackup/src/main/java/ratismal/drivebackup/uploaders/s3/S3Uploader.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+import static ratismal.drivebackup.config.Localization.intl;
+
+public class S3Uploader implements Uploader {
+    private UploadLogger logger;
+
+    public static String UPLOADER_NAME = "S3";
+    public static String UPLOADER_ID = "s3";
+
+    public MinioClient minioClient;
```
Comment (rhld16) [inline DriveBackup/src/main/java/ratismal/drivebackup/uploaders/s3/S3Uploader.java:34]: This client should be private

**Review by rhld16 — CHANGES_REQUESTED** [review]

**Review by lrmtheboss — APPROVED** [review]
LGTM just need to test now

**Comment by lrmtheboss** [comment]
Just tested with a local minio server, and worked fine

**Pull request merged**